### DISCUSSION TITLE
Content

### DISCUSSION THREAD

**Discussion by pervoj** [nahuelwexd/vala-website#12] [General]
Here we can try to put together the content of the website.

**Comment by dar5hak** [comment]
See also: https://github.com/colinkiama/vala-www/issues/1

**Comment by lw64** [comment]
I thought a bit about the content of the main page and have here a draft.
It is extra not touching really internals, like that it compiles to C, or that it uses GObject, because that won't interest anyone anyways.
Also the layout around it is what I have in mind, but doesn't need necessarily to be like that.
The big title and the "what can I build" section are advertising Vala good and describing what Vala is targeting at the moment (over time other things can be added when they are tested enough).
Also every use case should have a tutorial, so people can start easily.
With the "publish your library on ..." I have a website like crates.io in mind, where you can upload and browse GObject libraries (so you can write them and use them with Vala, but also other languages).
[vala-website.pdf](https://github.com/colinkiama/vala-www/files/8643313/vala-website.pdf)

**Reply by colinkiama** [reply]
I do agree that there should be much more content on the home page but I do think that the information ahould be spread out more. See: https://github.com/colinkiama/vala-www/issues/1#issuecomment-877693133

**Reply by nyx-lyb3ra** [reply]
While I like much of the content you have written, I feel we should not use “Why Vala?” here, but rather in the About.

**Reply by Suzie97** [reply]
I feel that we should ask the wider programming community about it so that we can get a wide range of opinions and then make a plan that satisfies the majority

**Reply by lw64** [reply]
Without the "why Vala" there isn't so much description of what vala actually makes interesting to consider. On the Rust main page they have it too. And on others as well.

**Reply by colinkiama** [reply]
Could probably summarise that in the home page and have a “read more” link to a page where “Why Vala?” is explained in detail

**Reply by colinkiama** [reply]
> I feel that we should ask the wider programming community about it so that we can get a wide range of opinions and then make a plan that satisfies the majority

Yes. How do we reach the wider programming community though?

Also, could this be done in parallel? If we do it in parallel, when we are ready to launch, we don’t have to wait for the research to be done to have some content on the page.

**Reply by lw64** [reply]
I think the Vala community is enough. And maybe selected gnome/elementary developers.

**Reply by lw64** [reply]
> Could probably summarise that in the home page and have a “read more” link to a page where “Why Vala?” is explained in detail

I don't think we should have so much subpages. I also have the why Vala section in bullet points, but I don't know if that is shorter and also it is more elegant as full sentences. For example good tooling is not really something that belongs to community, but in the sentence it makes sense.

**Reply by colinkiama** [reply]
But it would just link to the about page?

**Reply by Suzie97** [reply]
> Yes. How do we reach the wider programming community though?
> 
> 
> 
> Also, could this be done in parallel? If we do it in parallel, when we are ready to launch, we don’t have to wait for the research to be done to have some content on the page.

We can make a google form or something else with a sone questions and then send it to as many people as we can. I think we can get more responses if some influential people to promote the survey on their socials. Also, the vala community on discord is pretty big, so we can expect a fair amount of responses.

**Reply by Suzie97** [reply]
> Could probably summarise that in the home page and have a “read more” link to a page where “Why Vala?” is explained in detail

A blog post explaining the benefits of vala in detail seems more appropriate to me even though it's basically the same thing.

**Reply by lw64** [reply]
> > Yes. How do we reach the wider programming community though?
> > Also, could this be done in parallel? If we do it in parallel, when we are ready to launch, we don’t have to wait for the research to be done to have some content on the page.
> 
> We can make a google form or something else with a sone questions and then send it to as many people as we can. I think we can get more responses if some influential people to promote the survey on their socials. Also, the vala community on discord is pretty big, so we can expect a fair amount of responses.

Honestly I don't think we need to ask so much in the wider community. Some key persons are enough to get it right.



> > Could probably summarise that in the home page and have a “read more” link to a page where “Why Vala?” is explained in detail
> 
> A blog post explaining the benefits of vala in detail seems more appropriate to me even though it's basically the same thing.

It definitely needs to be on the website. You can put anything on a blog post, but this is the key featureset and the marketing strategy for vala.

**Reply by Suzie97** [reply]
> It definitely needs to be on the website. You can put anything on a blog post, but this is the key featureset and the marketing strategy for vala.

Yes, I agree that the summary of the key features should be there on the website. I meant instead of a read more button or something like that, we could link to a blog post
